### PR TITLE
Features/pipelines

### PR DESCRIPTION
**Pull request by kaaboeld**
HHT-1263 HHT-1262 HHT-1261

### CODE REVIEW

File: src/assets/styles/grid/_variables.scss [hunk 1]
```diff
@@ -1,29 +1,18 @@
...
-) !default;
-
-$gridColumns: 12 !default;
-$gridRowGap: 16px !default;
-$gridColumnGap: 16px !default;
-$columnPrefix: '.col' !default;
-$alignPrefix: '.align' !default;
-
-$alignments: start, end, center, stretch;
+$grid-columns: 12 !default;
```
Comment (Kagameow) [inline src/assets/styles/grid/_variables.scss:1]: looks like old grid were accidentally merged on the top of the new one

File: src/app/shared/menu-items/menu-items.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ export interface Menu {
   icon: string;
 }
 
-const MENU_ITEMS: Menu[] = [
+const MENUITEMS = [
```
Comment (Kagameow) [inline src/app/shared/menu-items/menu-items.ts:10]: Looks like some changes from the main were accidentally replaced with older code

File: src/app/ui-component/pipeline/pipeline-run/pipeline-run.component.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  }
+  getProgress(): IPipelineStatusType[] {
+    const list = this.types;
+    const status = this.data?.status;
+    const result: IPipelineStatusType[] = [];
+    let i = 0;
+    Object.keys(list).forEach((key) => {
+      const name = list[key as keyof typeof list];
+      const phase = status?.phase ?? 0;
+      const error = (phase == i && status?.error) ?? false;
```
Comment (Kagameow) [inline src/app/ui-component/pipeline/pipeline-run/pipeline-run.component.ts:82]: I'm not sure about `==` here

File: src/app/ui-component/pipeline/pipeline-run/pipeline-run.component.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      this.subscribes?.forEach((item) => item.unsubscribe());
+    }
+  }
+
+  tickTimer() {
+    if (this.data) {
+      const startAt = this.data?.startAt;
+      if (startAt) {
+        this.setDuration(startAt);
+        if (this.isRealtime) {
```
Comment (Kagameow) [inline src/app/ui-component/pipeline/pipeline-run/pipeline-run.component.ts:64]: Can this method be made less nested?

File: src/app/ui-component/pipeline/pipeline-detail/pipeline-detail.component.ts [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+  getStatus(id: number = 0) {
+    return ['Failed', 'Successfull'][id];
+  }
+
+  tickTimer() {
+    if (this.data) {
+      const startAt = this.data?.startAt;
+      if (startAt) {
+        this.setDuration(startAt);
+        if (this.isRealtime) {
```
Comment (Kagameow) [inline src/app/ui-component/pipeline/pipeline-detail/pipeline-detail.component.ts:156]: Can this method be made less nested?

**Review by Kagameow — CHANGES_REQUESTED** [review]

File: src/assets/styles/grid/_variables.scss (hunk 1 above)
Comment (kaaboeld) [reply src/assets/styles/grid/_variables.scss:1]: unused 
image

File: src/app/shared/menu-items/menu-items.ts (hunk 2 above)
Comment (kaaboeld) [reply src/app/shared/menu-items/menu-items.ts:10]: +

File: src/app/ui-component/pipeline/pipeline-detail/pipeline-detail.component.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+  editorOpts = {
+    theme: 'vs',
+    language: 'javascript',
+    readOnly: true,
+    automaticLayout: true,
+  };
+
+  actions: IActionItem[] = [
+    {
+      label: 'Retray',
```
Comment (Kagameow) [inline src/app/ui-component/pipeline/pipeline-detail/pipeline-detail.component.ts:58]: Typo here, should be `Retry`

**Review by Kagameow — APPROVED** [review]

**Pull request merged**